pkg/reconciler/cache/cachedresources/cachedresources_reconcile_replication.go

```diff
@@ -81,14 +87,24 @@ func (r *replication) reconcile(ctx context.Context, cachedResource *cachev1alph
 		// Local informer is based on the specific types we want to replicate.
 		local, err := r.discoveringDynamicKcpInformers.ClusterWithContext(ctx, cluster).ForResource(gvr)
 		if err != nil {
-			logger.Error(err, "Failed to get local informer for resource", "resource", gvr)
 			cancel()
+			if cachedResource.DeletionTimestamp != nil && !danglingResources {
```

Comment:
Isn't it enough that this is already checked and handled at https://github.com/kcp-dev/kcp/pull/3730/files#diff-9723dbe0e395b66baceb9f3ae33514b289f83e93d441b248c1018b01d40df25fR77 ?